### DISCUSSION TITLE
Is LiveCharts2 is supported in C# Markup?

### DISCUSSION THREAD

**Discussion by mpa74** [unoplatform/uno#15523] [Q&A] [answered]
Hello!
I use in my new project C# Markup and try to add CartesianChart from LiveCharts2.
CartesianChart is succesfully added, but i can`t set a Series because of error "Method, delegate or event is expected"

![2024-02-17_14-21-14](https://github.com/unoplatform/uno/assets/10248117/0b62a6e0-0e9d-4392-8cb6-0b34b4a46e39)
 
Is there any way to use LiveCharts?
thanks!

**Comment by sasakrsmanovic** [comment]
I've pinged @beto-rodriguez from Live charts about it. Also maybe @dansiegel  or @pictos have a hint?

**Comment by pictos** [comment]
@mpa74 can you share your project? That way we can take a look

**Reply by mpa74** [reply]
Of coarse -  https://github.com/unoplatform/uno/discussions/15523#discussioncomment-8535816

**Comment by dansiegel** [comment, marked as answer]
I don't believe there is a pre-generated package for LiveCharts2 that gives you the C# Markup extensions. But it's not a problem. You can simply add the following:

```cs
[assembly: GenerateMarkupForAssembly(typeof(CartesianChart))]
```

This will generate the extensions for you so that you can continue to use all of the types in the LiveCharts assembly with your C# Markup (not just the CartesianChart).

**Reply by beto-rodriguez** [reply]
That is an amazing feature of Uno 🎉

**Reply by mpa74** [reply]
Great feature! But now error is on compile time
I add this line and while compiling get warning messages like this "Uno.Extensions.Markup.Generators.IncrementalExtensionsGeneratorBase`1' threw an exception."
... and compiling fail because  "Series cannot used as method"

this is my repo if it helps: https://github.com/mpa74/MyForecasts

Error is on SecondPage.cs

**Reply by beto-rodriguez** [reply]
@mpa74  I think you must first remove the LiveCharts C# markup (`.Series(...)`), then try to compile your solution so the generator does its job, then try again and see if the generator created those methods for you.

**Reply by mpa74** [reply]
I'm sorry I didn't tell that I work under a Mac.
I try to run project under Windows and it works!
Thank you for your time. So, I mark the answer as solution

**Comment by pictos** [comment]
@mpa74 I did run on GTK and it just works,

![image](https://github.com/unoplatform/uno/assets/20712372/873aa584-f764-48a2-9475-70a0a41b0b96)

**Reply by mpa74** [reply]
Yes, looks like problem on MacOs only. Sorry. I should indicate my OS  earlier

**Reply by jeromelaban** [reply]
@mpa74 it is very curious that the build would fail only on macOS. Could you try building on the mac using `dotnet build -bl` and post the `msbuild.binlog` file here? Thanks!

**Reply by mpa74** [reply]
That's it - https://drive.google.com/drive/folders/1BVt7GVx5iHcTl5C_Ox-3oarDmhtcUUUS?usp=sharing
